### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by oleksandr-lysanets**
image
image
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by oleksandr-lysanets** [comment]
image

File: taxi/views.py [hunk 1]
```diff
@@ -87,3 +90,31 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
...
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverLicenseUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+
+
+@login_required()
+def add_or_delete_driver(request, pk):
```
Comment (olena-hul) [inline taxi/views.py]: Rewrite to class-based view

File: templates/taxi/driver_confirm_delete.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h1>Delete driver?</h1>
+  <form action="" method="post">
+    {% csrf_token %}
+
+    <input type="submit" value="Yes" class="btn btn-danger">
+  </form>
+{% endblock %}
```
Comment (olena-hul) [inline templates/taxi/driver_confirm_delete.html]: Add empty lines to the templates where missing

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: taxi/urls.py [hunk 3]
```diff
@@ -46,6 +49,22 @@
...
+        name="driver-delete"
+    ),
+    path(
+        "drivers/<int:pk>/update/",
+        DriverLicenseUpdateView.as_view(),
+        name="driver-update"
+    ),
+    path(
+        "drivers/<int:pk>/add_delete",
+        DriverDeleteView.as_view(),
```
Comment (olena-hul) [inline taxi/urls.py]: You should not reuse DriverDeleteView here

File: taxi/urls.py [hunk 4]
```diff
@@ -46,6 +49,22 @@
...
+        name="driver-delete"
+    ),
+    path(
+        "drivers/<int:pk>/update/",
+        DriverLicenseUpdateView.as_view(),
+        name="driver-update"
+    ),
+    path(
+        "drivers/<int:pk>/add_delete",
```
Comment (olena-hul) [inline taxi/urls.py]: Add `/` to the end of the url

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: taxi/urls.py [hunk 5]
```diff
@@ -46,6 +49,17 @@
...
+    path(
+        "drivers/<int:pk>/delete/",
+        DriverDeleteView.as_view(),
+        name="driver-delete"
+    ),
+    path(
+        "drivers/<int:pk>/update/",
+        DriverLicenseUpdateView.as_view(),
+        name="driver-update"
+    ),
```
Comment (olena-hul) [inline taxi/urls.py:63]: Add or delete view should still be present, but you should use separate class-based AddDeleteDriverView, not just DeleteDriverView

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**